pandaserver/dataservice/closer.py

```diff
@@ -0,0 +1,469 @@
+        Returns:
+            bool: True if all sub datasets are done, False otherwise.
+        """
+        tmp_log = LogWrapper(_logger,
+                             f"check_sub_datasets_in_jobset-{datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None).isoformat('/')}")
+        # skip already checked
+        if self.all_subscription_finished is not None:
+            return self.all_subscription_finished
+        # get consumers in the jobset
+        jobs = self.task_buffer.getOriginalConsumers(self.job.jediTaskID, self.job.jobsetID, self.job.panda_id)
+        checked_ds = set()
+        for job_spec in jobs:
+            # collect all sub datasets
+            sub_datasets = set()
+            for file_spec in job_spec.Files:
+                if file_spec.type == "output":
+                    sub_datasets.add(file_spec.destinationDBlock)
+            sub_datasets = sorted(sub_datasets)
+            if len(sub_datasets) > 0:
+                # use the first sub dataset
+                sub_dataset = sub_datasets[0]
+                # skip if already checked
+                if sub_dataset in checked_ds:
+                    continue
+                checked_ds.add(sub_dataset)
+                # count the number of unfinished
+                not_finish = self.task_buffer.countFilesWithMap({"destinationDBlock": sub_dataset, "status": "unknown"})
+                if not_finish != 0:
+                    tmp_log.debug(
+                        f"related sub dataset {sub_dataset} from {job_spec.PandaID} has {not_finish} unfinished files")
+                    self.all_subscription_finished = False
+                    break
+        if self.all_subscription_finished is None:
+            tmp_log.debug("all related sub datasets are done")
+            self.all_subscription_finished = True
+        return self.all_subscription_finished
+
+    def determine_final_status(self, destination_dispatch_block: str) -> str:
+        """
+        Determine the final status of a dispatch block.
+
+        Args:
+            destination_dispatch_block (str): The destination dispatch block.
+
+        Returns:
+            str: The final status.
+        """
+        if self.job.destinationSE == "local" and self.job.prodSourceLabel in ["user", "panda"]:
+            # close non-DQ2 destinationDBlock immediately
+            final_status = "closed"
+        elif self.job.lockedby == "jedi" and self.is_top_level_dataset(destination_dispatch_block):
+            # set it closed in order not to trigger DDM cleanup. It will be closed by JEDI
+            final_status = "closed"
+        elif self.job.prodSourceLabel in ["user"] and "--mergeOutput" in self.job.jobParameters and self.job.processingType != "usermerge":
+            # merge output files
+            if self.first_indv_dataset:
+                # set 'tobemerged' to only the first dataset to avoid triggering many Mergers for --individualOutDS
+                final_status = "tobemerged"
```

Comment:
@tmaeno please check if "tobemerged" and if first_indv_dataset is still needed

Reply:
I think the entire elif block 

 elif self.job.prodSourceLabel in ["user"] and "--mergeOutput" in self.job.jobParameters and self.job.processingType != "usermerge":

can be removed, so the variable is not required.